app/src/main/java/cz/muni/fi/pv256/movio2/uco_410434/ui/FilmListFragment.java

```diff
@@ -0,0 +1,185 @@
+        filmList.add(new Category("Old Classics"));
+        filmList.add(new Film("The Shining", 4.8f, R.mipmap.shining));
+        filmList.add(new Film("Casablanca", 3.4f, R.mipmap.casablanca));
+    }
+
+    @Override
+    public void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        context = getActivity().getApplicationContext();
+    }
+
+
+    @Override
+    public void onAttach(Context activity) {
+        super.onAttach(activity);
+
+        try {
+            filmSelectListener = (OnFilmSelectListener) activity;
+        } catch (ClassCastException e) {
+            Log.e(TAG, "Activity must implement OnFilmSelectListener", e);
+        }
+    }
+
+    @Override
+    public void onDetach() {
+        super.onDetach();
+
+        filmSelectListener = null; //Avoid leaking the Activity
+    }
+
+    @Nullable
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+        View view = inflater.inflate(R.layout.fragment_film_list, container, false);
+        fillListView(view);
+        return view;
+    }
+
+    private void fillListView(View view) {
+
+        filmListView = view.findViewById(R.id.listview_films);
+
+        LinearLayoutManager filmsLayoutManager = new LinearLayoutManager(context);
+        filmListView.setLayoutManager(filmsLayoutManager);
+
+        if (filmList != null && !filmList.isEmpty()) {
+            setAdapter(filmListView, filmList);
+        }
+    }
+
+    private void setAdapter(RecyclerView view, List<FilmsListItem> filmList) {
+        FilmsAdapter filmsAdapter = new FilmsAdapter(filmList.toArray(new FilmsListItem[0]));
+        view.setAdapter(filmsAdapter);
+    }
+
+    public interface OnFilmSelectListener {
+        void onFilmSelect(Film film);
+    }
+
+    public class FilmsAdapter extends RecyclerView.Adapter<FilmsAdapter.ViewHolder> {
+
+        private static final int FILM = 0, CATEGORY = 1, EMPTY = 2;
+
+        private FilmsListItem[] items;
+
+        public FilmsAdapter(FilmsListItem[] items) {
+            this.items = items;
+        }
+
+        @Override
+        public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
+            View v;
+            switch (viewType) {
+                case FILM:
+                    v = LayoutInflater.from(parent.getContext())
+                            .inflate(R.layout.film_list_film_item, parent, false);
+                    break;
+                case CATEGORY:
+                    v = LayoutInflater.from(parent.getContext())
+                            .inflate(R.layout.film_list_category_item, parent, false);
+                    break;
+                case EMPTY:
+                    v = LayoutInflater.from(parent.getContext())
+                            .inflate(R.layout.film_list_empty_item, parent, false);
+                    break;
+                default:
+                    return null;
```

Comment:
tu ti moze viewholder byt null, ale potom v dalsich metodach nerobis nullcheck na ViewHolder

Reply:
ha, pozeram ze v nasom projekte v praci to mame vlastne rovnako :)